File: JavaScript/arrayToMatrix.js
function convertArrayToMatrix(arr) {
    const matrix = [];
  
    // Extract column names and sort them in lexicographically ascending order
    const columns = Object.keys(arr[0]).sort();
    matrix.push(columns);
  
    // Extract values for each column
    arr.forEach(obj => {
      const row = [];
      columns.forEach(col => {
        row.push(obj[col]);
      });
      matrix.push(row);
    });
  
    return matrix;
  }
  
  const arr = [
    {"b": 1, "a": 2},
    {"b": 3, "a": 4}
  ];
  
  const matrix = convertArrayToMatrix(arr);
  console.log(matrix);
  

File: Python/array_to_matrix.py
def convert_array_to_matrix(arr):
    matrix = []

    # Extract column names
    columns = list(arr[0].keys())
    matrix.append(columns)

    # Extract values for each column
    for obj in arr:
        row = [obj[col] for col in columns]
        matrix.append(row)

    return matrix

arr = [
    {"b": 1, "a": 2},
    {"b": 3, "a": 4}
]

matrix = convert_array_to_matrix(arr)
print(matrix)
